narwhal/primary/src/certifier.rs

```diff
@@ -388,6 +383,7 @@ impl Certifier {
                     let signature_service = self.signature_service.clone();
                     let metrics = self.metrics.clone();
                     let network = self.network.clone();
+                    fail_point_async!("narwhal-delay");
```

Comment:
@aschran do you plan to add more delays/failures ? I say it would be interesting to add delays for the following cases as well:
* when waiting to receive votes for a proposed header
* on the certificate_fetcher when fetching the missing certificates
* when broadcasting/receiving a newly formed certificate (on the synchronizer it self)

Reply:
Yes, just wanted to get this initial set in. can def add more.

Note:
- There's already a failure point for all inbound RPCs, which at least partially covers many of these cases. but it's in a non-async part of the code so only tests failure and not delay.
- any await point in the code will already get randomly reordered in simtest, so it less valuable to add delays to places where there's already an await, as compared to places where there aren't currently awaits. also @andll mentioned plans to make simtest better able to test concurrency without having to create extra await points just for that purpose, so I didn't want to overdo it in that category.